deep.js

```diff
@@ -189,47 +189,116 @@ const deepAssignAdded = (target, ...sources) => {
 /**
  * Determines whether two objects are equal. Works on nested structures.
  * Work with all primitive types like Number, String, Big Int.
- * It also Object, Array, Date and Regex
  * It also works when nested structure contains object and array
- * @param {Object} obj1 can be either an object or array
- * @param {Object} obj2 can be either an object or array
+ * @param {Object} a can be either an object or array
+ * @param {Object} b can be either an object or array
  * @returns {Boolean}
  */
 const deepEqual = (a, b) => {
     if (a === b) {
         return true;
     }
 
-    if (Array.isArray(a)) {
-        if (!Array.isArray(b)) {
-            return false;
-        }
-
-        if (a.length !== b.length) {
-            return false;
-        }
+    if (Array.isArray(a) && Array.isArray(b)) {
+      if (a.length !== b.length) {
+        return false;
+      }
 
-        return a.every((value, i) => {
-            return deepEqual(value, b[i]);
-        });
-    }
-    if (isObject(a) && isObject(b)) {
-        const keysA = Object.keys(a);
-        const keysB = Object.keys(b);
-
-        return (
-            deepEqual(keysA, keysB) &&
-            keysA.every(key => {
-                return deepEqual(a[key], b[key]);
-            })
-        );
+      return a.every((value, index) => {
+        return deepEqual(value, b[index]);
+      });
     }
-};
+  if (isObject(a) && isObject(b)) {
+    const keysA = Object.keys(a);
+    const keysB = Object.keys(b);
 
-const deepEqualAdded = (a, b) => {
-    throw `not yet implemented`;
+    return (
+      deepEqual(keysA, keysB) &&
+      keysA.every(key => {
+        return deepEqual(a[key], b[key]);
+      })
+    );
+  }
+  return false;
 };
 
 const isObject = x => {
     return typeof x === `object` && x !== null;
 };
+
+const validateArray = (a,b) => {
```

Comment:
I found that I was doing the same array looping in multiple places. I suck with naming and I am happy to take suggestions for an alternate name.